capi/include/blazesym.h

```diff
@@ -420,6 +420,20 @@ typedef struct blaze_symbolizer_opts {
    * ensure compatibility in the presence of member additions.
    */
   size_t type_size;
+  /**
+   * Array of debug directories to search for split debug information.
+   *
+   * These directories will be consulted (in given order) when resolving
+   * debug links in binaries. By default and when this member is NULL,
+   * `/usr/lib/debug` and `/lib/debug/` will be searched. Setting an array
+   * here will overwrite these defaults, so make sure to include these
+   * directories as desired. The array has to be terminated by a NULL
+   * pointer.
+   *
+   * Note that the directory containing a symbolization source is always an
+   * implicit candidate target directory of the highest precedence.
+   */
+  const char *const *debug_dirs;
```

Comment:
instead of this "NULL terminator", I'd prefer to pass explicit number of elements as a separate property. Do you have a strong preference here or it was just an arbitrary choice? I find these NULL conventions in the array to be more error-prone and surprising, tbh.

Reply:
We can do that, but it felt awkward because we want to differentiate between
- `NULL` (use defaults)
and
- `{NULL}` (use no additional directories)

If we translate that to an explicit argument it would map to:
- ptr: `NULL`, len: `0`
- ptr: `!NULL` (but not really pointing to anything of substance), len: `0`

Reply:
Yeah, I'd do this ptr NULL or !NULL. I think it has a good and meaningful semantics. If ptr is NULL, that means parameters is not set (so use default value), while if it's !NULL (presumably still valid pointer) with size zero, that means it's an empty zero-element array (empty slice, if you will).

The benefit of this approach is that it maps naturally also to parameters where value (which in this case is `const char *`, and so it's fine in this case) is not a pointer. E.g., if you had some array of integers (enums, small structs, whatnot) as an argument. This NULL convention as last element will probably not work. While NULL vs !NULL+sz==0 is consistent and will work.

Reply:
Sounds good, changed.